### DISCUSSION TITLE
Termination criteria: ftol calculated on feasible individuals only

### DISCUSSION THREAD

**Discussion by Anto-314** [anyoptimization/pymoo#531] [Q&A] [answered]
Hello,

First of all, thank you very much for this great framework!

I would like to define a termination criteria which will stop the optimization when an objective space tolerance (ftol) calculated on feasible (= constraints respected) individuals only is reached.

I'm not sure of what is the best way to do it:

- DefaultSingleObjectiveTermination with ftol only (only_feas=True seen in the constructor)?
- RobustTermination(...)?

Could you please help me?

Thank you in advance.

**Comment by blankjul** [comment, marked as answer]
I would probably write a custom termination for this and re-use or copy code from existing pymoo terminations.

I have recently posted in #527 an example of how to design a custom termination and how to use it.
Does this work for you?

**Comment by Anto-314** [comment]
Thank you for your support and sorry for my late reply.

I will try to define a custom termination with this example and go back here if necessary.
